pkg/cli/advisory_create.go

```diff
@@ -147,6 +147,11 @@ newly created advisory and any other advisories for the same package.`,
 				AdvisoryDocs: advisoryCfgs,
 			}
 
+			req.VulnerabilityID, err = advisory.GenerateCGAID(req.Package, req.Aliases[0])
+			if err != nil {
+				return fmt.Errorf("unable to create advisory id: %w", err)
+			}
+
```

Comment:
I think this is a reasonable starting place. We may want to consider handling the ID generation _within_ the lib functions (e.g. `advisory.Create(...)`, called just below), to avoid placing burden on the caller and the risk of them getting this wrong. That would make functions like `advisory.Create` non-deterministic, which maybe is bad, maybe not.

And in either case, I think we'll want to add some check early on in those functions to ensure we got what we're expecting in the `req.VulnerabilityID` field — empty if we make the above change, or filled with a CGA ID if we don't.

Reply:
move to inside